test/metabase/cmd/rotate_encryption_key_test.clj

```diff
@@ -11,12 +10,27 @@
             [metabase.db.spec :as db.spec]
             [metabase.driver :as driver]
             [metabase.models :refer [Database Setting]]
-            [metabase.models.interface :as models]
+            [metabase.models.interface :as interface]
             [metabase.test :as mt]
             [metabase.test.data.interface :as tx]
             [metabase.util.encryption :as encrypt]
             [metabase.util.encryption-test :as eu]
-            [toucan.db :as db]))
+            [toucan.db :as db]
+            [toucan.models :as models]))
+
+(defn do-with-model-type
+  [mtype in-type-fns f]
+  (let [type-fns        (var-get #'models/type-fns)
+        before-type-fns @type-fns]
+    (swap! type-fns update mtype merge in-type-fns)
+    (try
+      (f)
+      (finally
+        (reset! type-fns before-type-fns)))))
+
+(defmacro with-model-type
+  [mtype type-fns & body]
+  `(do-with-model-type ~mtype ~type-fns (fn [] ~@body)))
```

Comment:
@camsaul This may be useful (and more appropriate) in Toucan. I'm not 100% happy with the name, though. Let me know if you'd like a PR. :slightly_smiling_face: